### DISCUSSION TITLE
security querying sessionLogin capabilities

### DISCUSSION THREAD

**Discussion by centeljon** [petrleocompel/hikaxpro_hacs#104] [General]
one thing i'd like to ask

are you sending user/password combination in cleartext to axpro ?
i found this:

    hikax.py    
        def get_session_params(self):
            q_user = urllib.parse.quote(self.username)
            q_password = urllib.parse.quote(self.password)
    --->      response = requests.get(f"http://{q_user}:{q_password}@{self.host}{consts.Endpoints.Session_Capabilities}{q_user}")  <---

i think querying session login capabilities does not need a login like http://user:password@host
makes no sense to use basic auth here and then digest to finally login securely.

could you please take a look if i am right or misinterpreting this code.

regards,
jon

**Comment by petrleocompel** [comment]
Hi, you are looking at internal lib. Which is deprecated and in default state is deactivated. 
API to AXPro is used via https://github.com/petrleocompel/hikaxpro

And that part is not in Lib. 
https://github.com/petrleocompel/hikaxpro/blob/0ed7e62360a92fa086000a254f4399810c5805d8/src/hikaxpro.py#L36

It was considered for some systems to use basic auth but it is not used!
